src/bp/ui-studio/src/web/views/FlowBuilder/nodeProps/ConditionModalForm.jsx

```diff
@@ -185,8 +243,146 @@ export default class ConditionModalForm extends Component {
     )
   }
 
-  onConditionChanged = event => {
-    this.setState({ condition: event.target.value })
+  changeConditionType = event => {
+    const conditionType = event.target.value
+
+    if (conditionType === 'always') {
+      this.setState({ conditionType, condition: 'true' })
+    } else if (conditionType === 'intent') {
+      this.setState({ conditionType, condition: `event.nlu.intent.name === ''` })
+    } else {
+      this.setState({ conditionType })
+    }
+  }
+
+  handlePropsTypeChanged = option => this.setState({ matchPropsType: option }, this.updatePropertyMatch)
+  handlePropsFieldNameChanged = e => this.setState({ matchPropsFieldName: e.target.value }, this.updatePropertyMatch)
+  handlePropsExpressionChanged = e => this.setState({ matchPropsExpression: e.target.value }, this.updatePropertyMatch)
+  handleConditionChanged = e => this.setState({ condition: e.target.value })
+
+  handleMatchIntentChanged = option => {
+    this.setState({
+      matchIntent: option,
+      condition: `event.nlu.intent.name === '${option.value}'`
+    })
+  }
+
+  updatePropertyMatch() {
+    const { matchPropsType, matchPropsFieldName, matchPropsExpression } = this.state
+
+    if (matchPropsType && matchPropsFieldName && matchPropsExpression) {
+      this.setState({
+        condition: `${matchPropsType.value}.${matchPropsFieldName} ${matchPropsExpression}`
+      })
+    }
+  }
+
+  renderIntentPicker() {
+    if (!this.state.intents) {
+      return null
+    }
+
+    const intents = this.state.intents.map(({ name }) => ({ label: name, value: name }))
+    return (
+      <Select
+        name="matchIntent"
+        value={this.state.matchIntent}
```

Comment:
Could just be computed on the fly from the condition instead of being stored in the state

Reply:
Not sure to understand, the state is used to display the correct option in the select box